src/main/java/com/aws/greengrass/componentmanager/plugins/docker/exceptions/DockerPullException.java

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.componentmanager.plugins.docker.exceptions;
+
+import static com.aws.greengrass.deployment.errorcode.DeploymentErrorCode.DOCKER_PULL_ERROR;
+
+public class DockerPullException extends DockerImageDownloadException {
+    static final long serialVersionUID = -3387516993124229948L;
+
+    public DockerPullException(String message) {
+        super(message);
+        super.getErrorCodes().add(DOCKER_PULL_ERROR);
```

Comment:
nit - why not just have a super class method: addErrorCode that does this. or even super constructor that takes a the error code as a param and adds it along with the initial one

Reply:
We discussed offline. Super constructor would not be efficient. Will wrap with an `addErrorCode`.